src/redis-default.js

```diff
@@ -0,0 +1,104 @@
+import { promisify } from 'util'
+
+class RedisDefaultStore {
+  constructor (client, options = {}) {
+    const invalidClientError = new TypeError(
+      'First parameter must be a valid RedisClient instance.'
+    )
+
+    try {
+      if (client.constructor.name !== 'RedisClient') {
+        throw invalidClientError
+      }
+    } catch (err) {
+      throw invalidClientError
+    }
+    this.client = client
+    this.prefix = options.prefix || 'axios-cache'
+    this.maxScanCount = options.maxScanCount || 1000
+    this.getAsync = promisify(client.get).bind(client)
+    this.psetexAsync = promisify(client.psetex).bind(client)
+    this.delAsync = promisify(client.del).bind(client)
+    this.scanAsync = promisify(client.scan).bind(client)
+  }
+
+  calculateTTL (value) {
+    const now = Date.now()
+
+    if (value.expires && value.expires > now) {
+      return value.expires - now
+    }
+
+    // If there is no expires in value or the provided expire is before the current time
+
+    return -1
+  }
+
+  transformKey (key) {
+    return this.prefix + '_' + key
+  }
+
+  async getItem (key) {
+    const item = (await this.getAsync(this.transformKey(key))) || null
```

Comment:
doesn't get itself return `null` when the key is not to be found?

Reply:
This is to ensure, even if Redis returns undefined, we need to make sure the `JSON.parse` does not throw an error
Because `JSON.parse(undefined)` will throw error but `JSON.parse(null)` returns `null`